File: programmers/level-1/12912. 두 정수 사이의 합/solution.js
// https://programmers.co.kr/learn/courses/30/lessons/12912

const solution = (a, b) => {
    let m = Math.max(a, b), n = m === a ? b : a, ret = 0;
    for (; n <= m; n++) ret += n;
    return ret;
};

// better way
const add = (a, b) => a + b * (Math.abs(b - a)+1) / 2;

File: programmers/level-1/12912. 두 정수 사이의 합/solution.cc
#include <iostream>
#include <string>
#include <vector>

using namespace std;

long long solution(int a, int b) {
    return (long long)(a + b) * (abs(b - a) + 1) / 2;
}

int main() {
    cout << solution(3, 5);
    return 0;
}